packages/cross_file/lib/src/implementations/blob_x_file.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:web/web.dart';
+
+import '../web_helpers/web_helpers.dart';
+import '../x_file.dart';
+
+/// A CrossFile that works on web, backed by a [Blob].
+class BlobXFile implements XFile {
+  /// Construct an [XFile] from a [Blob].
+  ///
+  /// `name` needs to be passed from the outside, since it's only available
+  /// while handling [html.File]s (when the ObjectUrl is created).
+  BlobXFile(
+    Blob blob, {
+    String? mimeType,
+    String? displayName,
+    DateTime? lastModified,
+  })  : _browserBlob = blob,
+        _mimeType = mimeType,
+        _lastModified = lastModified ?? DateTime.fromMillisecondsSinceEpoch(0),
+        _name = displayName;
+
+  /// Creates a [XFile] from a web [File].
+  factory BlobXFile.fromFile(File file) {
+    return BlobXFile(
+      file,
+      mimeType: file.type,
+      displayName: file.name,
+      lastModified: DateTime.fromMillisecondsSinceEpoch(file.lastModified),
+    );
+  }
+
+  /// Construct an [XFile] from the [objectUrl] of a [Blob].
+  ///
+  /// Important: the Object URL of a blob must have been created by the same JS
+  /// thread that is attempting to retrieve it. Otherwise, the blob will be null.
+  ///
+  /// See: https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL_static
+  static Future<BlobXFile> fromObjectURL(
+    String objectUrl, {
+    String? mimeType,
+    String? displayName,
+    DateTime? lastModified,
+  }) async {
+    return BlobXFile(
+      await fetchBlob(objectUrl),
```

Comment:
"ObjectURL" files could be made more specialized if we keep the objectURL as the information until the data is needed, and THEN we `fetchBlob`. This is a similar case as `.fromPath` from the native side.

Reply:
Done.